tests/common/tensor_test.py

```diff
@@ -305,3 +308,41 @@ def test_weighted_sum_handles_uneven_higher_order_input(self):
                                   attention_array[0, i, j, 1] * sentence_array[0, 1])
                 numpy.testing.assert_almost_equal(aggregated_array[0, i, j], expected_array,
                                                   decimal=5)
+
+    def test_viterbi_decode(self):
```

Comment:
It'd be nice to have a test against a simple hand-worked example, where you're actually testing the scores, too.